blob/azureblob/example_test.go

```diff
@@ -42,232 +39,71 @@ func Example() {
-	}
-
-	// IMPORTANT: The calls below are intended to show how to use As() to obtain the Azure Blob Storage SDK types.
-	// Due to the fake credentials used above, below calls are not executed.
-
-	// Use Reader.As to obtain SDK type azblob.DownloadResponse.
-	// See https://godoc.org/github.com/Azure/azure-storage-blob-go/azblob#DownloadResponse for more information.
-	var nativeReader azblob.DownloadResponse
-	if r.As(&nativeReader) {
-	}
-
-	// Use Attribute.As to obtain SDK type azblob.BlobGetPropertiesResponse.
-	// See https://godoc.org/github.com/Azure/azure-storage-blob-go/azblob#BlobGetPropertiesResponse for more information.
-	var nativeAttrs azblob.BlobGetPropertiesResponse
-	attr, _ := b.Attributes(ctx, "key")
-	if attr.As(&nativeAttrs) {
-	}
-
-	// Use Bucket.As to obtain SDK type azblob.ContainerURL.
-	// See https://godoc.org/github.com/Azure/azure-storage-blob-go/azblob for more information.
-	var nativeBucket *azblob.ContainerURL
-	if b.As(&nativeBucket) {
-	}
-
-	// Use WriterOptions.BeforeWrite to obtain SDK type azblob.UploadStreamToBlockBlobOptions.
-	// See https://godoc.org/github.com/Azure/azure-storage-blob-go/azblob#UploadStreamToBlockBlobOptions for more information.
-	beforeWrite := func(as func(i interface{}) bool) error {
-		var nativeWriteOptions *azblob.UploadStreamToBlockBlobOptions
-		if as(&nativeWriteOptions) {
-		}
-		return nil
-	}
-	wopts := &blob.WriterOptions{
-		ContentType: "application/json",
-		BeforeWrite: beforeWrite,
-	}
-	// Create a *blob.Writer.
-	w, _ := b.NewWriter(ctx, "key", wopts)
-	w.Write([]byte("{"))
-	w.Write([]byte(" message: 'hello' "))
-	w.Write([]byte("}"))
-	w.Close()
-
-	// Use ListOptions.BeforeList to obtain SDK type azblob.ListBlobsSegmentOptions.
-	// See https://godoc.org/github.com/Azure/azure-storage-blob-go/azblob#ListBlobsSegmentOptions for more information.
-	beforeList := func(as func(i interface{}) bool) error {
-		var nativeListOptions *azblob.ListBlobsSegmentOptions
-		if as(&nativeListOptions) {
-		}
-		return nil
-	}
-
-	// Iterate through a virtual directory.
-	iter := b.List(&blob.ListOptions{Prefix: "blob-for-delimiters", Delimiter: "/", BeforeList: beforeList})
-	for {
-		if p, err := iter.Next(ctx); err == io.EOF {
-			break
-		} else if err == nil {
-			if p.IsDir {
-				// Use ListObject.As to obtain SDK type azblob.BlobPrefix.
-				// See https://godoc.org/github.com/Azure/azure-storage-blob-go/azblob#BlobPrefix for more information.
-				var nativeDirObj azblob.BlobPrefix
-				if p.As(&nativeDirObj) {
-				}
-			} else {
-				// Use ListObject.As to obtain SDK type azblob.BlobItem.
-				// See https://godoc.org/github.com/Azure/azure-storage-blob-go/azblob#BlobItem for more information.
-				var nativeBlobObj azblob.BlobItem
-				if p.As(&nativeBlobObj) {
-				}
-			}
-		}
-	}
-
-	// Output:
-	// ReadAll failed due to invalid credentials
+	b, err = blob.Open(ctx, "azblob://mycontainer?cred_path=replace-with-path-to-credentials-file")
+	_ = b
+	_ = err
```

Comment:
nit: err can be moved before `_ = b` or ignored entirely.

Reply:
Sorry, I didn't get this. I need to touch both b and err or else there's a compile error that they aren't used.